crypto/x509/x509_vfy.c

```diff
@@ -2980,7 +2984,14 @@ static int build_chain(X509_STORE_CTX *ctx)
                  * certificate among the ones from the trust store.
                  */
                 if ((search & S_DOALTERNATE) != 0) {
-                    OPENSSL_assert(num > i && i > 0 && ss == 0);
+                    if (!ossl_assert(num > i && i > 0 && ss == 0)) {
+                        X509err(X509_F_BUILD_CHAIN, ERR_R_INTERNAL_ERROR);
+                        X509_free(xtmp);
+                        trust = X509_TRUST_REJECTED;
+                        ctx->error = X509_V_ERR_UNSPECIFIED;
+                        search = 0;
+                        continue;
```

Comment:
> It isn't clear to me that is that trivial.

What I meant was that logic in beginning of switch after "/* last chance */" section is so trivial that it can be replicated elsewhere, which might be more readable. This is the key, readability. In a sense your comment kind of confirms what I'm trying to say. The code is not trivial, and it would be a win if it was more readable/transparent.

> I would be uncomfortable making a potentially significant change

It wasn't my intention to suggest to do it right away.